pilot/pkg/model/sidecar.go

```diff
@@ -77,6 +86,9 @@ type SidecarScope struct {
 	// destination rule.
 	destinationRules map[host.Name]*Config
 
+	// Set of all destinationRules this sidecar depends on.
+	destinationRuleDependencies map[string]struct{}
```

Comment:
Why do you need this when you have destinationRules above?

Reply:
the destinationRules field above is a mapping of service hostName to relevant config rather than "destinationRule name" to that and thus does not meet our requirement as the key in PushRequest(or XdsEvent).configsUpdate is config name itself.